spec/services/migration/migrators/contract_spec.rb

```diff
@@ -0,0 +1,111 @@
+require "rails_helper"
+
+RSpec.describe Migration::Migrators::Contract do
+  it_behaves_like "a migrator", :contract, %i[cohort lead_provider course statement] do
+    def create_ecf_resource
+      cohort = create(:ecf_migration_cohort)
+      npq_lead_provider = create(:ecf_migration_npq_lead_provider)
+      course = create(:ecf_migration_npq_course)
+      version = rand(1..9)
+
+      create(
+        :ecf_migration_statement,
+        cpd_lead_provider: npq_lead_provider.cpd_lead_provider,
+        cohort:,
+        contract_version: "0.0.#{version}",
+      )
+      create(
+        :ecf_migration_npq_contract,
+        npq_lead_provider:,
+        cohort:,
+        course_identifier: course.identifier,
+        version: "0.0.#{version}",
+        created_at: rand(1..100).days.ago,
+        updated_at: rand(1..100).days.ago,
+      )
+    end
+
+    def create_npq_resource(ecf_resource)
+      lead_provider = create(:lead_provider, ecf_id: ecf_resource.npq_lead_provider.id)
+      cohort = create(:cohort, start_year: ecf_resource.cohort.start_year)
+      course = create(:course, identifier: ecf_resource.course_identifier)
+      ecf_statement = Migration::Ecf::Finance::Statement.where(cpd_lead_provider: ecf_resource.npq_lead_provider.cpd_lead_provider).first!
+      statement = create(
+        :statement,
+        ecf_id: ecf_statement.id,
+        lead_provider:,
+        cohort:,
+      )
+      contract_template = create(
+        :contract_template,
+        ecf_id: ecf_resource.id,
+      )
+      create(
+        :contract,
+        statement:,
+        course:,
+        contract_template:,
+      )
+    end
+
+    def setup_failure_state
+      cohort = create(:ecf_migration_cohort)
+      npq_lead_provider = create(:ecf_migration_npq_lead_provider)
+
+      create(
+        :ecf_migration_statement,
+        cpd_lead_provider: npq_lead_provider.cpd_lead_provider,
+        cohort:,
+        contract_version: "0.0.1",
+      )
+
+      create(
+        :ecf_migration_npq_contract,
+        npq_lead_provider:,
+        cohort:,
+        version: "0.0.1",
+      )
+    end
+
+    describe "#call" do
+      it "sets the created Contract and ContractTemplate attributes correctly" do
```

Comment:
might be worth using the shared context here for migrators

Reply:
I think it already is?